src/windows/wslaservice/exe/WSLAContainer.cpp

```diff
@@ -69,9 +69,40 @@ HRESULT WSLAContainer::Start()
 }
 
 HRESULT WSLAContainer::Stop(int Signal, ULONG TimeoutMs)
+try
 {
-    return E_NOTIMPL;
+    std::lock_guard lock{m_lock};
+
+    /* 'nerdctl stop ...'
+     *   returns success and <containerId> on stdout if the container is running or already stopped
+     *   returns error "No such container: <containerId>" on stderr if the container is in 'Created' state or does not exist
+     *
+     * For our case, we consider stopping a container that is not running or does not exist as a no-op and return success.
+     * TODO: Discuss and return stdout/stderr or corresponding HRESULT from nerdctl stop for better diagnostics.
+     */
+
+    if (m_state == WslaContainerStateExited)
```

Comment:
We should call `State()` here so the state is updated if the init process has returned

Reply:
Easiest, maybe, but not the most efficient. recursive_mutexes typically have more overhead. I've added a StateNoLock(), or the other option would be to pass a bool into State() to indicate if lock is already held. I don't have a strong preference for either of the latter approaches, but would like to avoid recursive_mutex if possible